src/prusti/all.ml

```diff
@@ -16,9 +15,36 @@ let rec all xs pred
    all(xs, p, res) ==> ens res=all(xs, p)
 @*)
 
+let rec repeat x n =
+  if n <= 0 then []
+  else x :: repeat x (n - 1)
+
+let is_positive x = x > 0
+
+let rec all_positive xs =
+  match xs with
+  | [] -> true
+  | x :: xs' -> x > 0 && all_positive xs'
+
+(* let has_property p xs = all p xs *)
+
+let test_all_positive n
+(*@ ex r ys; all_positive(ys, r); req res=ys; ens r=true/\res=ys @*)
+= repeat 1 n
+
 (* Unlike pure length, this is not provable because p on the left may have effects *)
 
-let test1 xs
+(* let test1 xs (* FIXME *)
```

Comment:
Previously, I wrongly thought that these tests pass if we require `xs=[1;2;3;4]` instead of `xs->[1;2;3;4]` (https://github.com/EmilyOng/AlgebraicEffect/issues/19)